packages/sw-precaching/src/lib/revisioned-cache-manager.js

```diff
@@ -0,0 +1,66 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import assert from '../../../../lib/assert';
+import installHandler from './install-handler';
+
+class RevisionedCacheManager {
+  constructor() {
+    this._eventsRegistered = false;
+    this._cachedAssets = [];
+  }
+
+  cache({revisionedFiles} = {}) {
+    assert.isInstance({revisionedFiles: revisionedFiles}, Array);
+
+    const parsedFileList = revisionedFiles.map((revisionedFile) => {
+      let parsedAssetHash = revisionedFile;
+      if (typeof revisionedFile === 'string') {
+        parsedAssetHash = {path: revisionedFile, revision: revisionedFile};
+      }
+
+      // TODO Check revisionedFile.path and revisionedFile.revision
+
+      if (parsedAssetHash.path.indexOf('http') !== 0) {
+        // Must be a relative path
+        parsedAssetHash.path =
+          new URL(parsedAssetHash.path, location.origin).toString();
+      }
+
+      return parsedAssetHash;
+    });
+
+    this._cachedAssets = this._cachedAssets.concat(parsedFileList);
+
+    this._registerEvents();
```

Comment:
There are two options for this, adding the install event regardless of whether anything is cached or not OR adding the install event listener when we know it's going to be used for something.

Sounds like you are opting for the former by putting it in the constructor.

However we can't tell if the developer will actually use this module until they call cache - which is *after* the constructor.